coalib/bearlib/languages/documentation/DocumentationComment.py

```diff
@@ -33,3 +41,107 @@ def __init__(self, documentation, language, docstyle,
 
     def __str__(self):
         return self.documentation
+
+    def parse_documentation(self):
+        """
+        Parses documentation independent of language and docstyle.
+
+        :return:
+            A list of the parsed documentation metadata.
+        :raises:
+            NotImplementedError
+        """
+        if self.language == "python":
+            return self.parse_python()
+        else:
+            raise NotImplementedError(
+                "Documentation parsing for {} in {}"
+                " has not been implemented yet".format(
+                    self.language, self.docstyle))
+
+    def parse_python(self):
+        """
+        Parse documentation. Usable attributes are:
+
+        ``:param``
+        ``@param``
+
+        ``:return``
+        ``@return``
+
+        :return: A list of parsed sections(descriptions, params, return values)
+        """
+        lines = self.documentation.splitlines(keepends=True)
+
+        param_identifiers = (":param ", "@param ")
+        return_identifiers = (":return: ", "@return: ")
+
+        return self._parse_documentation(lines, param_identifiers,
+                                         return_identifiers)
+
+    def _parse_documentation(self, lines, params, returns):
+        """
+        Parses documentation based on parameter and return symbols.
+
+        :param lines:   The lines of documentation to parse.
+        :param params:  The string with which a parameter description starts.
+        :param returns: The string with which a return description starts.
+
+        :return:        The list of all the parsed sections of the
+                        documentation.
+        """
+        parse_mode = self._ParseMode.DESCRIPTION
+
+        cur_param = ""
+
+        desc = ""
+        retval_desc = ""
+
+        parsed = list()
+
+        for line in lines:
+
+            if line.strip().startswith(params):
+                parse_mode = self._ParseMode.PARAM
+                splitted = line[7:].split(": ", 1)
+                cur_param = splitted[0].strip()
+                param_desc = ""
+                # For cases where the param description is not on the
+                # same line, but on subsequent lines.
+                try:
+                    param_desc = splitted[1]
+                except IndexError:
+                    pass
+                parsed.append(self.Param(name=cur_param, desc=param_desc))
+
+                continue
+
+            if line.strip().startswith(returns):
```

Comment:
when removing `concat_doc_parts`, you can also turn this one into an `elif` and the one below too, then you can remove the `continue` statement :+1: